### PR TITLE
HOSTEDCP-1483: update cloud-credential-operator in go.mod

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**: update cloud-credential-operator to newer version that doesn't use a deprecated import with a CVE

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [HOSTEDCP-1483](https://issues.redhat.com/browse/HOSTEDCP-1483)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -68,23 +68,23 @@ require (
...
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.30.1
-	k8s.io/apiextensions-apiserver v0.30.1
-	k8s.io/apimachinery v0.30.1
-	k8s.io/apiserver v0.30.1
+	k8s.io/api v0.30.2
+	k8s.io/apiextensions-apiserver v0.30.2
+	k8s.io/apimachinery v0.30.2
+	k8s.io/apiserver v0.30.2
 	k8s.io/cli-runtime v0.30.1
```
Comment (bryan-cox) [inline go.mod:83]: We should bump the other k8s to the same version, cli-runtime, kube-scheduler, kubectl, pod-security-admission.

You'll probably need to bump the replaces at the bottom as well.

Comment (Patryk-Stefanski) [reply go.mod]: seems like we do need it after all, bumped two of the versions there is no kubernetes v0.30.2 so I'm leaving it as is.

**Review by devguyio — COMMENTED** [review]
/lgtm

**Pull request merged**